docs/user-guide/pydantic-schema-manager.rst

```diff
@@ -0,0 +1,226 @@
+
+Using the manager
+=================
+
+Before you can serialize model instances, you need to register the model class with the manager.
+Models must be subclasses of `AvroBaseModel`_ from `dataclasses-avroschema`_. Then, you can serialize instances of the model into Avro messages with the registry schema ID embeded in them.
+
+.. code-block:: python
+
+   import asyncio
+
+   from dataclasses_avroschema.pydantic import AvroBaseModel
+   from safir.schema_manager import (
+       SchemaRegistryConnectionSettings,
+       PydanticSchemaManager,
+   )
+   from schema_registry.client import AsyncSchemaRegistryClient
+
+
+   async def main() -> None:
+       # Construct a manager
+       config = SchemaRegistryConnectionSettings()
+       registry = AsyncSchemaRegistryClient(
+           **config.schema_registry_client_params
+       )
+       manager = PydanticSchemaManager(registry=registry)
+
+       # Define a model
+       class MyModel(AvroBaseModel):
+           field1: int = Field(default=0)
+           field2: str
+
+       # Register your model, which will:
+       #
+       # * Create the schema in the registry if it doesn't exist
+       # * Create a new version of the schema in the registry if this model has
+       #   changed since #   the last time it was registered
+       # * Do nothing if the schema already exists in the registry and the
+       #   model hasn't changed.
+       await manager.register(MyModel)
+
+       # Serialize an instance
+       instance = MyModel(field1=1, field2="woohoo!")
+       avro: bytes = manager.serialize(instance)
+
+
+   asyncio.run(main())
+
+You probably want to register all of your models at the start of your application so that:
+
+* Any schema registry lookups are cached before your app starts doing stuff
+* Any incompatible changes you have made to the models will prevent the app from starting
+
+.. _AvroBasemodel: https://marcosschroh.github.io/dataclasses-avroschema/pydantic/
+.. _dataclasses-avroschema: https://github.com/marcosschroh/dataclasses-avroschema
+
+Compatibility
+-------------
+
+One of the main benefits of using the schema registry is that it can tell you if you try to update a schema in a way that is incompatible with how other users of the schema are expecting it to be updated.
+The specific changes that are incompatible could vary from subject to subject, and depend on that subject's `compatibility type`_.
+
+You can specify the compatibility type when registering a model.
+If you don't specify a compatibility type, the subject will have the default compatibility type set on the schema registry server.
+
+Once the initial version of the schema is created, if you change the model in your app in an incompatible way and try to register it again, the manager will throw an ``IncompatibleSchemaError``.
+
+.. code-block:: python
+
+   from safir.schema_manager import (
+       IncompatibleSchemaError,
+       SchemaRegistryCompatibility,
```

Comment:
I'm actually totally with you on this, I was trying to match style elsewhere in the repo, like https://github.com/lsst-sqre/safir/blob/main/safir/src/safir/github/models.py

Reply:
I have a whole complicated set of mental heuristics for this that are probably too complicated to write down and thus are probably unrealistically complicated, heh. Yeah, I'd probably use the longer names for the GitHub models because classes like that feel different to me. I think the distinction is how much context there is at the call site. Class names feel like they're often used in situations with less context, and also they're often multi-line blocks anyway so the penalty for longer names is less. Enums are often used as arguments to functions or methods where the context is really obvious, so I tend to make them shorter. This is absolutely me overthinking all of this.

Reply:
I shortened some other names in the kafka connection stuff too:
https://github.com/lsst-sqre/safir/pull/303/commits/617b8daa65454ed9b16622cf05917f2719db2587